File: java/prob3.java
public class prob3
{
	public static void main(String[] args)
	{
		long num = 600851475143L;
		double sqrt = Math.sqrt(num);
		for(long i = 2;i < sqrt; i++)
		{
			if( num % i == 0 && IsPrime(i))
			{
				System.out.println("Prime Factor: "+i);
			}
		}
	}

	public static boolean IsPrime(long num)
	{
		double sqrt = Math.sqrt(num);
		for(long i=2;i<sqrt;i++)
		{
			if(num % i == 0)
			{
				return false;
			}
		}
		return true;
	}
}

File: haskell/prob3.hs
main = putStrLn ("All factors: " ++ show (allPrimeFactors 600851475143))

isPrime :: Integer -> Bool
isPrime k = null [ x | x <- [2..k - 1], k `mod`x  == 0]

-- k = possible factor
-- x = the number of factor
isFactor :: Integer -> Integer -> Bool
isFactor k x = x `mod`k  == 0

allPrimeFactors :: Integer -> [Integer]
allPrimeFactors num = [ x | x <- [1..(floor.sqrt.fromIntegral)(num)] , (isFactor x num) && isPrime x ] 